Sources/TuistGenerator/Generator/ProjectGenerator.swift

```diff
@@ -78,13 +79,25 @@ final class ProjectGenerator: ProjectGenerating {
 
         let xcodeprojPath = sourceRootPath.appending(component: "\(project.fileName).xcodeproj")
 
+        // Project and workspace.
+        return try generateProjectAndWorkspace(project: project,
+                                               graph: graph,
+                                               sourceRootPath: sourceRootPath,
+                                               xcodeprojPath: xcodeprojPath)
+    }
+
+    // MARK: - Fileprivate
+
+    private func generateProjectAndWorkspace(project: Project,
+                                             graph: Graphing,
+                                             sourceRootPath: AbsolutePath,
+                                             xcodeprojPath: AbsolutePath) throws -> GeneratedProject {
         // Derived files
         let deleteOldDerivedFiles = try derivedFileGenerator.generate(project: project, sourceRootPath: sourceRootPath)
 
-        // Project and workspace.
         let workspaceData = XCWorkspaceData(children: [])
         let workspace = XCWorkspace(data: workspaceData)
-        let projectConstants = determineProjectConstants()
+        let projectConstants = try determineProjectConstants()
         let pbxproj = PBXProj(objectVersion: projectConstants.objectVersion,
```

Comment:
What about adding a test that ensures that the `PBXProj`'s objectVersion and archiveVersion have the right values depending on the returned constant?

Reply:
Added 👍